### PR TITLE
Adding alternative configs for Piantor and update example

### PR DESCRIPTION
**Pull request by duncanam**
## Description

Here we add an entrypoint for the Weact controller which is an option that Beekeeb ships with their Piantor kits, as well as the required variable to define which vbus pin the vanilla Pico uses vs. the Weact Pico. Additionally, added an example flashing with the Weact config as well as an alternative layout, Miryoku (this was originally puzzling for me, and this would have been helpful).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* enable Weact for Piantor
* add example
* adding "no vbus sense" for microcontrollers without that mode assembled in the Piantor

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage). **(I'm currently typing this using this change)**

### CODE REVIEW

**Comment by duncanam** [comment]
@l4u Mind reviewing?

**Comment by dunk2k** [comment]
In `readme.md` could you add a paragraph with conditions for when to compile no_vbus_sense?

**Comment by duncanam** [comment]
Sure thing, good idea.

-------- Original Message --------

File: keyboards/beekeeb/piantor/no_vbus_sense/config.h [hunk 1]
```diff
@@ -0,0 +1 @@
+#undef USB_VBUS_PIN
```
Comment (drashna) [inline keyboards/beekeeb/piantor/no_vbus_sense/config.h:1]:
```suggestion
// Copyright 2023 QKM
// SPDX-License-Identifier: GPL-2.0-or-later
#pragma once

#undef USB_VBUS_PIN
```

File: keyboards/beekeeb/piantor/weact/config.h [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#undef USB_VBUS_PIN
```
Comment (drashna) [inline keyboards/beekeeb/piantor/weact/config.h:1]:
```suggestion
// Copyright 2023 QKM
// SPDX-License-Identifier: GPL-2.0-or-later
#pragma once

#undef USB_VBUS_PIN
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by duncanam** [comment]
@dunk2k and/or @drashna - what is this repo's policy on squashing commits during PR? Would you like me to add commits with edits then squash at the end, squash as I add, or don't squash at all and leave a full paper trail (albeit messier) upon merge with master?

**Comment by drashna** [comment]
> @dunk2k and/or @drashna - what is this repo's policy on squashing commits during PR? Would you like me to add commits with edits then squash at the end, squash as I add, or don't squash at all and leave a full paper trail (albeit messier) upon merge with master?

We squash when we merge PRs.  So the commit history from the PR doesn't really matter.

Though, personally, I prefer that it's not "cleaned up", since it makes it easier to see when and where changes occurred. But that's just my preference.

**Comment by duncanam** [comment]
TODO for me:

- [ ] add no vbus sense example
- [ ] update with copyright header

**Review by dunk2k — APPROVED** [review]

**Comment by duncanam** [comment]
Build failing on bcat targets here- I may need some guidance on how best to proceed to fix.

**Comment by dunk2k** [comment]
Thankfully I've alrqady

> Build failing on bcat targets here- I may need some guidance on how best to proceed to fix.

Thankfully I've already a PR (#21205) for fixing that. 
Regardless, as it's a community/userspace keymap, it wouldn't be your remit/responsibility to fix 😉

File: keyboards/beekeeb/piantor/no_vbus_sense/config.h [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// Copyright 2023 beekeeb
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#undef USB_VBUS_PIN
```
Comment (drashna) [inline keyboards/beekeeb/piantor/no_vbus_sense/config.h:4]:
```suggestion
#pragma once

#undef USB_VBUS_PIN
```

File: keyboards/beekeeb/piantor/weact/config.h [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// Copyright 2023 beekeeb
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#undef USB_VBUS_PIN
```
Comment (drashna) [inline keyboards/beekeeb/piantor/weact/config.h:4]:
```suggestion
#pragma once

#undef USB_VBUS_PIN
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by duncanam** [comment]
@l4u just wanted to also check in with you on this one too

**Comment by duncanam** [comment]
@drashna Any other suggestions here?

**Review by drashna — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
This should really be inverted, that way there is no need to undef. And keyboard updates need to go through `develop`.

**Comment by duncanam** [comment]
@fauxpark How do we invert such that the default config has that pin defined but the two special configs have variations (either a new pin in the case of the Weact microcontroller, or no pin, for that case)?

**Comment by fauxpark** [comment]
Probably the easiest would be to split the board into two "revisions" instead of how you've done it here. Then make sure the "base" cannot be built, either by removing the rules.mk or setting `DEFAULT_FOLDER` to the revision corresponding to the current config.

**Comment by duncanam** [comment]
@fauxpark Okay let's see if I am understanding this: we can remove the pin definition from the current "base" config and place this into a new folder, "base", with its own rules, and define a new constant DEFAULT_FOLDER in the root definition that points to "base", such that when building the config it will default to " base" if not specified?

**Comment by fauxpark** [comment]
Basically. You'd have two subfolders named say `pico` and `weact`, both with empty rules.mk (to make them build targets), but one also has a config.h with the VBUS pin define. Then you add `DEFAULT_FOLDER = beekeeb/piantor/pico` to the existing `rules.mk`.

**Comment by duncanam** [comment]
> Basically. You'd have two subfolders named say `pico` and `weact`, both with empty rules.mk (to make them build targets), but one also has a config.h with the VBUS pin define. Then you add `DEFAULT_FOLDER = beekeeb/piantor/pico` to the existing `rules.mk`.

Got it- that does seem cleaner.

File: keyboards/beekeeb/piantor/config.h [hunk 5]
```diff
@@ -10,3 +10,6 @@
 #define SERIAL_USART_FULL_DUPLEX
 #define SERIAL_USART_TX_PIN GP0
 #define SERIAL_USART_RX_PIN GP1
+
+// When USB_VBUS_PIN is not defined, SPLIT_USB_DETECT is used.
+#define USB_VBUS_PIN GP24 // for Raspberry Pi Pico
```
Comment (drashna) [inline keyboards/beekeeb/piantor/config.h:15]:
```suggestion
```

File: keyboards/beekeeb/piantor/no_vbus_sense/config.h [hunk 6]
```diff
@@ -0,0 +1,5 @@
+// Copyright 2023 beekeeb
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#undef USB_VBUS_PIN
```
Comment (drashna) [inline keyboards/beekeeb/piantor/no_vbus_sense/config.h:5]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by duncanam** [comment]
Thanks for the feedback, I'll take a look

**Comment by duncanam** [comment]
Status update: I am slow

**Comment by duncanam** [comment]
Currently verifying some changes on local hardware.

-------- Original Message --------

**Pull request closed without merging**